modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/Cluster/ClientClusterGroupProjection.cs

```diff
@@ -0,0 +1,126 @@
+﻿/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ * 
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Core.Impl.Client.Cluster
+{
+    using System;
+    using System.Collections.Generic;
+    using Apache.Ignite.Core.Binary;
+
+    /// <summary>
+    /// Projection builder that is used for remote nodes filtering.
+    /// </summary>
+    internal sealed class ClientClusterGroupProjection
+    {
+        /** */
+        private const int Attribute = 1;
+
+        /** */
+        private const int ServerNode = 2;
```

Comment:
Right now it seems to me that this approach should be simplified.
We could take care only about attributes pairs since most of the filters just rely on a specific attribute name.
Consider forServers() and "org.apache.ignite.cache.client".  This will dramaticaly simplify the filterer. 

In contrast, we can not use this approach to say for "forOlderst()" filter.